webapp/src/package.ts

```diff
@@ -764,6 +781,48 @@ export function getEditorPkg(p: pxt.Package): EditorPackage {
     return newOne
 }
 
+
+/**
+ * Fetches all packages that have a dependency on the given package
+ *
+ */
+export function getDependents(pkgid: string) {
+    const deps: string[] = []
+    Object.keys(mainPkg.deps).forEach(id => {
+        const pkg = mainPkg.deps[id]
+        if (pkg.dependencies(false)[pkgid]) {
+            deps.push(id)
+        }
+    })
+    return deps
+}
+
+export function findExtForNs(ns: string): pxt.Package {
+    const pkgId = Object.keys(mainPkg.deps).find( dep => {
+        const associatedNamespaces = getNsForPkg(dep)
+        return associatedNamespaces?.includes(ns)
+    })
+    return mainPkg.deps[pkgId]
+}
+
+export function getNsForPkg(pkgId: string) {
```

Comment:
how do you handle radio-broadcast who adds blocks to "radio"?

Reply:
There'll just be one x for the whole category, and it'll delete radio and radio-broadcast. I think the eventual plan is to change all the extensions that add blocks to an existing category to make their own (since I think we're the only ones who do that?)

Reply:
what if an extension add a block to "basic"? :)

Reply:
Do you know if any extensions do that? I think since basic isn't considered an extension to begin with, it won't be deleted (whereas radio is a package!)

Reply:
I don't think this is a concern that extensions add to basic. We can decide not approve such extensions.